lib/Toolbar/index.js

```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react';
+import icons from './icons';
+
+class Toolbar extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { showHeaders: false };
+  }
+
+  render() {
+    return (<div id="quill-toolbar">
+      <div className="ql-formats">
+        <button
+          className="ql-show-headers"
+          onClick={() => this.setState({ showHeaders: !this.state.showHeaders })}
+        >
+          { icons.aa() }
+        </button>
+        <div className={`ql-select-headers ${this.state.showHeaders ? 'show-headers' : ''}`}>
+          <button className="ql-paragraph" />
+          <button className="ql-header" value="1" />
+          <button className="ql-header" value="2" />
+          <button className="ql-header" value="3" />
```

Comment:
Just wondering if we should extract the toolbar "actions" as children rather than having them explicitly set in the toolbar component.

Having the UI component bound to `ql` means we store references to a 3rd party that doesn't exist in `gather-ui`. E.g. imagine seeing this component for the first time...what does `ql` stand for?

So when consuming you would;

```
<Toolbar id="some-id">
  <div className="ql-formats">
    <button className="ql-header" value="1" />
    <button className="ql-header" value="2" />
    <button className="ql-header" value="3" />
    ...etc
  </div>
</Toolbar>
```

If you need to then apply BEM classes to the children you could look into https://facebook.github.io/react/docs/react-api.html#react.children to map over and add additional markup with the BEM classes you need.

Reply:
The trouble with separating the logic here is it is almost all business/action based, there's not much UI in it. Arguably if we just had `<Toolbar id="some-id">...children components...</Toolbar>` as you suggest, well why do we have a specific new component and why don't we just use a `<div>`? Or maybe we just have a Toolbar component back in the App as it was originally?

Reply:
Well, the styling side of things is well worth keeping detached from the consuming app. The UI story would then demo the use of all the styles. 

Even the smallest component when handling UI logic or styling should be considered for the UI library, but again its just 2 paths ending at the same destination.

If we use the UI library I suggest we don't add 3rd party markup. Let's keep the concerns separate, otherwise when adding future components what is right or wrong? 😄

Reply:
There are some more complex examples that we could take for this component, which would make it truly extendable/flexible for the future but I believe they'd be too complex for our needs at this moment 😄